test/Tests/src/Data/Statistics_Spec.enso

```diff
@@ -156,6 +168,36 @@ spec =
             [].compute Statistic.Skew . is_nan . should_equal True
             [].compute Statistic.Kurtosis . is_nan . should_equal True
 
+    Test.group "Statistics - NaN Vector " <|
+        Test.specify "should be able to count and sum on NaN Vector" <|
+            [Number.nan, Number.nan].compute . should_equal 0
+            [Number.nan, Number.nan].compute Statistic.Sum . should_equal 0
+
+        Test.specify "should fail with Empty_Error for Minimum and Maximum on NaN Vector" <|
+            [Number.nan, Number.nan].compute Statistic.Minimum . should_fail_with Empty_Error
+            [Number.nan, Number.nan].compute Statistic.Maximum . should_fail_with Empty_Error
+
+        Test.specify "should be NaN for other statistics sum on NaN Vector" <|
+            [Number.nan, Number.nan, Number.nan, Number.nan].compute Statistic.Mean . is_nan . should_equal True
+            [Number.nan, Number.nan, Number.nan, Number.nan].compute Statistic.Variance . is_nan . should_equal True
+            [Number.nan, Number.nan, Number.nan, Number.nan].compute Statistic.Skew . is_nan . should_equal True
+            [Number.nan, Number.nan, Number.nan, Number.nan].compute Statistic.Kurtosis . is_nan . should_equal True
+
+    Test.group "Statistics - Nothing Vector " <|
+        Test.specify "should be able to count and sum on Nothing Vector" <|
+            [Nothing, Nothing].compute . should_equal 0
+            [Nothing, Nothing].compute Statistic.Sum . should_equal 0
+
+        Test.specify "should fail with Empty_Error for Minimum and Maximum on Nothing Vector" <|
+            [Nothing, Nothing].compute Statistic.Minimum . should_fail_with Empty_Error
+            [Nothing, Nothing].compute Statistic.Maximum . should_fail_with Empty_Error
+
+        Test.specify "should be NaN for other statistics sum on Nothing Vector" <|
+            [Nothing, Nothing, Nothing, Nothing].compute Statistic.Mean . is_nan . should_equal True
+            [Nothing, Nothing, Nothing, Nothing].compute Statistic.Variance . is_nan . should_equal True
+            [Nothing, Nothing, Nothing, Nothing].compute Statistic.Skew . is_nan . should_equal True
+            [Nothing, Nothing, Nothing, Nothing].compute Statistic.Kurtosis . is_nan . should_equal True
+
```

Comment:
Ideally we should check `running` for these cases as well.

Reply:
Also it seems like it may be worth checking vectors that start with a `Nothing` prefix and then have some values.

Reply:
Just checked on this branch:
```
from Standard.Base import all

main =
    IO.println <| [Nothing, Nothing, Nothing].running Statistic.Count
    IO.println <| [Nothing, Nothing, Nothing].running Statistic.Sum
    IO.println <| [Nothing, Nothing, Nothing].running Statistic.Variance
    IO.println <| [Nothing, Nothing, Nothing].running Statistic.Maximum
    
    IO.println <| [Nothing, Nothing, 10, 20, Nothing, 30].running Statistic.Count
    IO.println <| [Nothing, Nothing, 10, 20, Nothing, 30].running Statistic.Variance
    IO.println <| [Nothing, Nothing, 10, 20, Nothing, 30].running Statistic.Maximum
```
yields:
```
[0, 0, 0]
[0, 0, 0]
[NaN, NaN, NaN]
(Error: Empty_Error)
[0, 0, 1, 2, 2, 3]
[NaN, NaN, NaN, 50.0, 50.0, 100.0]
(Error: Empty_Error)
```

So it seems basically all cases work correctly apart from Min/Max. We need special handling for the `running` case.

If the vector is all nothing, I guess throwing empty error seems acceptable (although returning `[Nothing, Nothing, Nothing]` seems more intuitive). But if the vector _does_ have valid elements, just starts with nulls - here I don't think we should be throwing. But maybe I'm wrong?

My intuition says, last line should be `[Nothing, Nothing, 10, 20, 20, 30]`.